internal/server/application.go

```diff
@@ -0,0 +1,113 @@
+package server
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	coreruleset "github.com/corazawaf/coraza-coreruleset"
+	"github.com/corazawaf/coraza-coreruleset/io"
+	"github.com/corazawaf/coraza-spoa/internal/config"
+	"github.com/corazawaf/coraza/v3"
+	"github.com/corazawaf/coraza/v3/types"
+	"github.com/negasus/haproxy-spoe-go/message"
+	"github.com/rs/zerolog"
+	"github.com/yalue/merged_fs"
+)
+
+type ErrInterrupted struct {
+	Interruption *types.Interruption
+}
+
+func (e ErrInterrupted) Error() string {
+	return fmt.Sprintf("interrupted with status %d and action %s", e.Interruption.Status, e.Interruption.Action)
+}
+
+func (e *ErrInterrupted) Is(target error) bool {
```

Comment:
I think this is wrong. `Is` is not meant to see if two errors are the same, instead check error wrapping.

Reply:
I'm playing around with errors.As, apparently that's a shortcut to compare and cast at the same time.
And it is working